src/matrix/matrixnormal.jl

```diff
@@ -0,0 +1,161 @@
+MatrixNormal(M::AbstractMatrix, U::AbstractPDMat,          V::LinearAlgebra.Cholesky) = MatrixNormal(M, U,        PDMat(V))
+
+MatrixNormal(m::Int, n::Int) = MatrixNormal(zeros(m, n), Matrix(1.0I, m, m), Matrix(1.0I, n, n))
+
+#  -----------------------------------------------------------------------------
+#  REPL display
+#  -----------------------------------------------------------------------------
+
+show(io::IO, d::MatrixNormal) = show_multline(io, d, [(:M, d.M), (:U, Matrix(d.U)), (:V, Matrix(d.V))])
+
+#  -----------------------------------------------------------------------------
+#  Conversion
+#  -----------------------------------------------------------------------------
+
+function convert(::Type{MatrixNormal{T}}, d::MatrixNormal) where T <: Real
+    MM = convert(AbstractArray{T}, d.M)
+    UU = convert(AbstractArray{T}, d.U)
+    VV = convert(AbstractArray{T}, d.V)
+    MatrixNormal{T, typeof(MM), typeof(UU)}(MM, UU, VV, T(d.c0))
+end
+
+function convert(::Type{MatrixNormal{T}}, M::AbstractMatrix, U::AbstractPDMat, V::AbstractPDMat, c0) where T <: Real
+    MM = convert(AbstractArray{T}, M)
+    UU = convert(AbstractArray{T}, U)
+    VV = convert(AbstractArray{T}, V)
+    MatrixNormal{T, typeof(MM), typeof(UU)}(MM, UU, VV, T(c0))
+end
+
+#  -----------------------------------------------------------------------------
+#  Properties
+#  -----------------------------------------------------------------------------
+
+size(d::MatrixNormal) = size(d.M)
+
+rank(d::MatrixNormal) = minimum( size(d) )
+
+insupport(d::MatrixNormal, X::Matrix) = isreal(X) && size(X) == size(d)
+
+mean(d::MatrixNormal) = d.M
+
+mode(d::MatrixNormal) = d.M
+
+params(d::MatrixNormal) = (d.M, d.U, d.V)
+
+@inline partype(d::MatrixNormal{T}) where {T<:Real} = T
+
+#  -----------------------------------------------------------------------------
+#  Evaluation
+#  -----------------------------------------------------------------------------
+
+function _matrixnormal_c₀(U::AbstractPDMat, V::AbstractPDMat)
+
+    n = dim(U)
+    p = dim(V)
+
+    -(n * p / 2) * (logtwo + logπ) - (n / 2) * logdet(V) - (p / 2) * logdet(U)
+
+end
+
+function logkernel(d::MatrixNormal, X::AbstractMatrix)
+
+    A  = X - d.M
+    At = Matrix(A')
+
+    -0.5 * tr( (d.V \ At) * (d.U \ A) )
+
+end
+
+_logpdf(d::MatrixNormal, X::AbstractMatrix) = logkernel(d, X) + d.c0
+
+#  -----------------------------------------------------------------------------
+#  Sampling
+#  -----------------------------------------------------------------------------
+
+function _rand!(rng::AbstractRNG, d::MatrixNormal, A::AbstractMatrix)
+
+    n, p = size(d)
+
+    X = randn(rng, n, p)
+
+    A .= d.M + d.U.chol.L * X * d.V.chol.U
+
+end
+
+#  -----------------------------------------------------------------------------
+#  Transformation
+#  -----------------------------------------------------------------------------
+
+vec(d::MatrixNormal) = MvNormal(vec(d.M), kron(d.V.mat, d.U.mat))
```

Comment:
Does this use of `vec` have any precedent? 
I wonder if this should instead just be a `MvNormal` constructor?
i.e. ` MvNormal(d::MatrixNormal) = MvNormal(vec(d.M), kron(d.V.mat, d.U.mat))`

Reply:
Right. Given #307, I don't think there's agreement on this in general.

Reply:
I will put a vote in for this being a `MvNormal` constructor (I also see no reason not to have both `vec` and `MvNormal` do this) -- but i will leave it to you / maintainers to decide :)

Reply:
I think I am ok with vec, but you need to document this behaviour and add it to the doc (you can add specific methods to the doc, it's already done